### PR TITLE
[Fix](function) remove datev2 signature of microsecond

### PR DESCRIPTION
**Pull request by zclllyybb**
## Proposed changes

Issue Number: close #xxx

for sql
```sql
select microsecond(d) from dates; -- d is Date type
```
before:
```sql
ERROR 1105 (HY000): errCode = 2, detailMessage = [INTERNAL_ERROR]Function microsecond get failed, expr is VectorizedFnCall[microsecond](arguments=d,return=Nullable(Int32)) and return type is Nullable(Int32).
```
now: cast argument to `Date` and get result `0`

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zclllyybb** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/Microsecond.java [hunk 1]
```diff
@@ -50,6 +51,13 @@ public Microsecond(Expression arg) {
         super("microsecond", arg);
     }
 
+    @Override
+    public void checkLegalityBeforeTypeCoercion() {
+        if (!getArgument(0).getDataType().isDateTimeV2Type()) {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/Microsecond.java:56]: if only support datetimev2 type, remove datev2 signature is just ok

Comment (zclllyybb) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/Microsecond.java]: got it. fixed~

**Comment by zclllyybb** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Review by yiguolei — APPROVED** [review]

**Comment by yiguolei** [comment]
run buildall

**Review by yiguolei — APPROVED** [review]

**Pull request merged**